src/snowflake/connector/converter_snowsql.py

```diff
@@ -198,6 +198,13 @@ def _TIMESTAMP_NTZ_to_python(self, ctx: dict[str, Any]) -> Callable:
         def conv(value: str) -> str:
             microseconds, fraction_of_nanoseconds = _extract_timestamp(value, ctx)
             try:
+                new_integer_part = str(microseconds).split(".")[0]
+                origin_integer_part = value.split(".")[0]
+                if new_integer_part != origin_integer_part and float(
+                    origin_integer_part
+                ) + 1 == float(new_integer_part):
```

Comment:
Maybe the correct behavior for "off by more than 1" should be failing, or is there legit use case? Not sure if it is going to be a behavior change though. Ideally we should not hit this issue or we are already silently corrupting the data I think?

Reply:
unfortunately we're already silently corrupting customer's data.

this is a legit use case and a limitation of the Python float number -- from what I learn  Python float 64-bit double-precision values, and once the integer part is great than a certain value, it will sacrifice and round the decimal part.

e.g. if you type `123456789123456.99999` in the REPL, python returns `123456789123457.0`

> We observed this issue from the customer who uses the max possible timestamp value 9999-12-31 23.59.59.99999999 in their table, the value queried on SnowSQL ouputs 10000-01-01 00:00:00.999999.